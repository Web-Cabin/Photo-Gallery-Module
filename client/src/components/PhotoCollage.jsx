import React from 'react';
import styled from 'styled-components';

const ImageOne = styled.img`
    display: inline-block;
    left: 0%;
    position: absolute;
    transform: scale(1);
    opacity: 1;
    transition: -ms-transform 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s, -webkit-transform 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s, transform 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s, opacity 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s;
    height: 456px;
    width: 684px; 
    box-sizing: border-box;
    cursor: pointer;
    -webkit-transition: all .5s ease;
    -moz-transition: all .5s ease;
    -ms-transition: all .5s ease;
    -o-transition: all .5s ease;
    transition: all .5s ease;

    ${ImageOne}:hover {
        transform: scale(0.9);
        box-shadow: 0 0 10px rgba(0, 0, 0, 0.5);
        -webkit-transform: scale(1.2);
        -moz-transform: scale(1.2);
        -o-transform: scale(1.2);
        -ms-transform: scale(1.2); 
        transform: scale(1.2)
    }
`;

const ContainerOne = styled.div`
    &:hover ${ImageOne} {
        display: none;
    }
`;



const ImageTwo = styled.img`
    display: inline-block;
    position: absolute;
    transform: scale(1);
    left: 50%;
    opacity: 1;
    transition: -ms-transform 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s, -webkit-transform 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s, transform 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s, opacity 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s;
    height: 227.333px;
    width: 341px; 
    box-sizing: border-box;
    cursor: pointer;
`;

const ImageThree = styled.img`
    display: inline-block;
    position: absolute;
    transform: scale(1);
    right: 0%;
    opacity: 1;
    transition: -ms-transform 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s, -webkit-transform 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s, transform 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s, opacity 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s;
    height: 227.333px;
    width: 341px; 
    box-sizing: border-box;
    cursor: pointer;
`;

const ImageFour = styled.img`
    display: inline-block;
    position: absolute;
    transform: scale(1);
    left: 50%;
    top: 31%;
    opacity: 1;
    transition: -ms-transform 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s, -webkit-transform 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s, transform 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s, opacity 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s;
    height: 227.333px;
    width: 341px; 
    box-sizing: border-box;
    cursor: pointer;
`;

const ImageFive = styled.img`
    display: inline-block;
    position: absolute;
    transform: scale(1);
    right: 0%;
    top: 31%;
    opacity: 1;
    transition: -ms-transform 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s, -webkit-transform 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s, transform 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s, opacity 450ms cubic-bezier(0.645, 0.045, 0.355, 1) 0s;
    height: 227.333px;
    width: 341px; 
    box-sizing: border-box;
    cursor: pointer;
`;

const PhotoCollage = ({ photos, renderCarousel }) => {
    return (
        <div>
        <ImageOne src={`${photos[0]}`} onClick={() => renderCarousel()}/>
        <ImageTwo src={`${photos[1]}`} onClick={() => renderCarousel()} />
        <ImageThree src={`${photos[2]}`} onClick={() => renderCarousel()} />
        <ImageFour src={`${photos[3]}`} onClick={() => renderCarousel()} />
        <ImageFive src={`${photos[4]}`} onClick={() => renderCarousel()} />
        </div>
    );
}

export default PhotoCollage;
