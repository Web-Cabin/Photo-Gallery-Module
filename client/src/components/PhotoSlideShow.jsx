import React from 'react';
import styled from 'styled-components';


const SliderWrapper = styled.div`
  position: relative;
  height: 100%;
  width: 100%;
`;


export const BackArrow = styled.div`
height: 50px;
width: 50px;
display: flex;
align-items: center;
justify-content: center;
background: #f9f9f9;
cursor: pointer;
transition: transform ease-in .1s;
position: absolute;
top: 46%;
left: 60px;
z-index: 999;
color: #fff;
background-image: url('https://photogalleryproject.s3.us-east-2.amazonaws.com/backarrow.png');
`;

export const NextArrow = styled.div`
height: 50px;
width: 50px;
display: flex;
align-items: center;
justify-content: center;
background: #f9f9f9;
border-radius: 50%;
cursor: pointer;
transition: transform ease-in .1s;
position: absolute;
top: 46%;
right: 435px;
z-index: 999;
color: #fff;
background-image: url('https://photogalleryproject.s3.us-east-2.amazonaws.com/nextarrow.png');
`;


export const Slides = styled.div`
display: inline-block;
height: 525px;
width: 785px;
top: 30px;
border-radius: 15px;
`;


class PhotoSlideShow extends React.Component {
  constructor(props) {
      super(props);
      
      this.state = {
        translateValue: 0
      }
  }
  
    render() {
      console.log('rendering photoslideshow');
      return (
        <div>
          <SliderWrapper style={{
              transform: `translateX(${this.props.translateValue}px)`,
              transition: 'transform ease 0.25s'
            }}> 
              {
                this.props.photos.map((photo, i) => (
                  <Slide key={i} photo={photo} />
                ))
              }
          </SliderWrapper>
            
         <LeftArrow goToPrevSlide={this.props.goToPrevSlide} index={this.props.index} handleClick={this.props.handleClick.bind(this)}/>

         <RightArrow goToNextSlide={this.props.goToNextSlide} index={this.props.index} handleClick={this.props.handleClick.bind(this)}/> 

      </div>
      );
    }
  }
  


export const Slide = ({ photo }) => {
    var styles = {
        backgroundImage: `url(${photo})`,
        backgroundSize:'cover',
        backgroundRepeat: 'no-repeat',
        backgroundPosition: '50% 60%'
    }
    return (
        <Slides className="slide" style={styles}></Slides>
    );
  };

  
export const LeftArrow = ({ handleClick, index }) => {
    return (
      <BackArrow className="back-arrow" onClick={() => handleClick(index - 1)} >
      </BackArrow>
    );
  }
  
  
export const RightArrow = ({ handleClick, index }) => {
    return (
      <NextArrow className="next-arrow"  onClick={() => handleClick(index + 1)} >
      </NextArrow>
    );
  }


export default PhotoSlideShow;



