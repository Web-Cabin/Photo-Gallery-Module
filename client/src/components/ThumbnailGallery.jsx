import React from 'react';

const ThumbnailWrapper = styled.div`
    position: relative; 
    flex-direction:row;
    height: 100%;
    width: 100%; 
    top: 10%;
    right: 5%;
`;

export const Thumbnail = styled.div`
    display: inline-flex;
    height: 58px;
    width: 58px;
    border-radius: 5px;
    margin: 7px;
    padding: 5px;
`;



class ThumbnailGallery extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      borderColor: false, 
    }
  }


  render() {
    return (
      <div>
           <ThumbnailWrapper style={{
               transform: `translateX(${this.props.translateValue}px)`,
               transition: 'transform ease-out 0.45s'
            }}>
          {
              this.props.photos.map((image, index) => (
              <Thumbnails image={image} borderColor={this.state.borderColor} key={index} id="thumbnailId" handleClick={this.props.handleClick.bind(this, index)} />
            ))
          }
          </ThumbnailWrapper>
      </div>
    );
  }
}


export const Thumbnails = ({ image, handleClick, borderColor }) => {
    var styles = {
        backgroundImage: `url(${image})`,
        backgroundSize:'cover',
        backgroundRepeat: 'no-repeat',
        backgroundPosition: 'top 50% right 50%',
    }
    return (
        <Thumbnail className="thumbnail" style={styles} onClick={(event) => handleClick(event)} ></Thumbnail>
    );
}

export default ThumbnailGallery; 