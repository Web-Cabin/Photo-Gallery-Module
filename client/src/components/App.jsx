import React from 'react';
import PhotoCollage from './PhotoCollage.jsx';
import axios from 'axios';
import $ from 'jquery';
import PhotoSlideShow from './PhotoSlideShow.jsx';

class App extends React.Component {
    
    constructor(props) {
        super(props);

        this.state = {
            photos: [],
            showSlideshow: false,
        }
        this.fetchPhotos = this.fetchPhotos.bind(this);
        this.renderCarousel = this.renderCarousel.bind(this);
    }

    componentDidMount() {
        this.fetchPhotos();
    }
    
    fetchPhotos() {
        $.ajax({
            url: '/api/displayphotos/8',
            method: 'GET',
            success: (response) => {
                // array.toString().split(',');
                console.log(response);
                response = response[0].photo_url.toString().split(',');
                this.setState({
                    photos: response,
                });
            },
            error: (data) => {
                console.log('failure to retrieve', data); 
            }
        })
    }

    renderCarousel() {
        this.setState({
            showSlideshow: true,
        });
    }


    render() {

        return (
            <div className="container">
                {this.state.showSlideshow? <PhotoSlideShow photos={this.state.photos} /> : <PhotoCollage photos={this.state.photos} renderCarousel={this.renderCarousel}/>}
            </div>
        );
    }
}

export default App; 