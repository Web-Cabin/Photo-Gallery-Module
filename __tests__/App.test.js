import React from 'react';
import { shallow, mount, render } from 'enzyme';
import PhotoCollage from '../client/src/components/PhotoCollage.jsx';
import PhotoSlideShow, { DescriptionList, BackArrow, NextArrow, LeftArrow, RightArrow } from '../client/src/components/PhotoSlideShow.jsx';
import App from '../client/src/components/PhotoCollage.jsx';
import ThumbnailGallery, { Thumbnails, Thumbnail } from '../client/src/components/ThumbnailGallery.jsx';
import { wrap } from 'module';


//photo collage should render 5 images onto the page 
describe('<PhotoCollage />', () => {
    it('renders five images', () => {
        const photos = ['https://photogalleryproject.s3.us-east-2.amazonaws.com/image1.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image2.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image3.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image4.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image5.jpeg'];

        const wrapper = shallow(<PhotoCollage photos={photos} />);

        expect(wrapper.find('.image')).toHaveLength(5);
    });

    it('tests click event', () => {
        const photos = ['https://photogalleryproject.s3.us-east-2.amazonaws.com/image1.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image2.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image3.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image4.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image5.jpeg'];

        const button1 = shallow((<PhotoCollage photos={photos} renderCarousel={() => 1}></PhotoCollage>));
        button1.find('.image').at(0).simulate('click')
        expect(button1).toHaveLength(1);

        const button2 = shallow((<PhotoCollage photos={photos} renderCarousel={() => 1}></PhotoCollage>));
        button2.find('.image').at(1).simulate('click')
        expect(button2).toHaveLength(1);

        const button3 = shallow((<PhotoCollage photos={photos} renderCarousel={() => 1}></PhotoCollage>));
        button3.find('.image').at(2).simulate('click')
        expect(button3).toHaveLength(1);

        const button4 = shallow((<PhotoCollage photos={photos} renderCarousel={() => 1}></PhotoCollage>));
        button4.find('.image').at(3).simulate('click')
        expect(button4).toHaveLength(1);

        const button5 = shallow((<PhotoCollage photos={photos} renderCarousel={() => 1}></PhotoCollage>));
        button5.find('.image').at(4).simulate('click')
        expect(button5).toHaveLength(1);
    });
});
     

//check for buttons 
describe('<PhotoSlideShow />', () => {
    const photos = ['https://photogalleryproject.s3.us-east-2.amazonaws.com/image1.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image2.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image3.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image4.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image5.jpeg'];

  const wrapper = shallow(<PhotoSlideShow photos={photos} />);
  it ('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
  });

  const descriptions = ['this is a test', 'this is the room', 'this is the bathroom', 'this is the balcony'];

  const description = shallow(<DescriptionList descriptions={descriptions}/>);
  it ('renders correctly', () => {
      expect(description).toMatchSnapshot();
  });

  it('tests click events for back arrow', () => {
      const mockCallBack = jest.fn();

      const button = shallow((<LeftArrow onClick={mockCallBack}></LeftArrow>));
      button.find('.back-arrow').simulate('click');
      expect(button).toHaveLength(1);
  })

  it('tests click events for next arrow', () => {
      const mockCB = jest.fn();

      const nextButton = shallow((<RightArrow onClick={mockCB}></RightArrow>));
      nextButton.find('.next-arrow').simulate('click');
      expect(nextButton).toHaveLength(1);
  });
});

describe('<ThumbnailGallery />', () => {
    const photos = ['https://photogalleryproject.s3.us-east-2.amazonaws.com/image1.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image2.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image3.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image4.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image5.jpeg'];

    let wrapper = shallow(<ThumbnailGallery photos={photos} />);
    it ('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it ('test click event', () => {
        const mockCallback = jest.fn();

        const button = shallow((<Thumbnails handleClick={mockCallback}></Thumbnails>));
        button.find('.thumbnail').simulate('click');
        expect(mockCallback).toHaveBeenCalled();
    });

    it ('tests click event to see if it triggers a component method to run', () => {
        const mockCallback = jest.fn();
        const photos = ['https://photogalleryproject.s3.us-east-2.amazonaws.com/image1.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image2.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image3.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image4.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image5.jpeg'];


        const wrap = mount((<ThumbnailGallery photos={photos} ></ThumbnailGallery>));
        wrap.find('.thumbnail').at(1).simulate('click');
        expect(wrap.state('currentThumbnailIndex')).toBe(0);
        expect(wrap.state('translateValue')).toBeGreaterThanOrEqual(0);
    });

    it ('tests state change', () => {
        const photos = ['https://photogalleryproject.s3.us-east-2.amazonaws.com/image1.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image2.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image3.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image4.jpeg', 'https://photogalleryproject.s3.us-east-2.amazonaws.com/image5.jpeg'];

        let wrap = shallow(<ThumbnailGallery photos={photos}/>);
        expect(wrap.state('currentThumbnailIndex')).toBeGreaterThanOrEqual(0);
        expect(wrap.state('translateValue')).toBeGreaterThanOrEqual(0);
    });
});












//1) test descriptions are on page 
//2) test that clicks work
//3) test that props are changing on clicks 

// describe('<App />', () => {
//     it('should change the state componentState componentDidMount method is invoked', () => {
//         expect(wrapper.state('componentState')).toEqual('mounted');
//     });
// });


// describe('<Share />', () => {
//     it('should call function when button is clicked', () => {
//         // const button = shallow(<Button name='share-button' handleClick={func} />)
//         button.simulate('click');
//         expect(func).toHaveBeenCalled();
//     });

//     it('should render correctly', () => {
//         // const button = shallow(<Button name='share-button' />);
//         expect(button).toMatchSnapshot();
//     });
// })


// //view photos should be clickable 
// describe('<ViewPhotos />', () => {
//     it('should call function when button is clicked', () => {
//         // const button = shallow(<Button name='view-photos-button' handleClick={func} />)
//         button.simulate('click');
//         expect(func).toHaveBeenCalled();
//     });

//     it('should render correctly', () => {
//         // const button = shallow(<Button name='view-photos-button' />);
//         expect(button).toMatchSnapshot();
//     });
// })
// //view photos should be clickable 

// //App should render 6 components 