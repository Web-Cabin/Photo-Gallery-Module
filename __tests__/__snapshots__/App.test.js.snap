// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoSlideShow /> renders correctly 1`] = `ShallowWrapper {}`;

exports[`<ThumbnailGallery /> renders correctly 1`] = `ShallowWrapper {}`;
